presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -301,6 +336,26 @@ private synchronized void refreshNodesInternal()
         // nodes by connector id changes anytime a node adds or removes a connector (note: this is not part of the listener system)
         activeNodesByConnectorId = byConnectorIdBuilder.build();
 
+        if (isMemoizeDeadNodesEnabled) {
+            SetView<String> deadNodeIds = difference(
```

Comment:
Do we need to initialize nodes outside of this method? Otherwise how does this work?

Reply:
There is `nodes` that is local to `refreshNodesInternal` and we also have `this.nodes` that is global variable. We calculate the local `nodes` and then make the global `nodes` a copy of this local one.

Reply:
Ah got it, I was wondering how it was working, lol.
Can we rename one of them to make sure its more clear to read?

Reply:
We do the same for `AllNodes` , i think it is fine to leave as it.

Reply:
sure, sounds good with me.